pkg/reconciler/pipelinerun/pipelinerun_test.go

```diff
@@ -835,6 +836,72 @@ func TestUpdateTaskRunsState(t *testing.T) {
 
 }
 
+func TestUpdateTaskRunsWithWhenExpressions(t *testing.T) {
+	// TestUpdateTaskRunsWithWhenExpressions runs "getTaskRunsStatus" and verifies how it updates a PipelineRun status
+	// from a TaskRun with When Expressionsassociated to the PipelineRun
+	pr := tb.PipelineRun("test-pipeline-run", tb.PipelineRunNamespace("foo"), tb.PipelineRunSpec("test-pipeline"))
+	pipelineTask := v1beta1.PipelineTask{
+		Name:    "unit-test-1",
+		TaskRef: &v1beta1.TaskRef{Name: "unit-test-task"},
+		WhenExpressions: []v1beta1.WhenExpression{{
+			Input:    "foo",
+			Operator: selection.NotIn,
+			Values:   []string{"foo", "bar"},
+		}},
+	}
+	task := tb.Task("unit-test-task", tb.TaskSpec(
+		tb.TaskResources(tb.TaskResourcesInput("workspace", resourcev1alpha1.PipelineResourceTypeGit)),
+	), tb.TaskNamespace("foo"))
+
+	expectedWhenExpressionsResults := &v1beta1.WhenExpressionsEvaluationStatus{
+		EvaluationResults: []*v1beta1.WhenExpressionEvaluationResult{{
+			Expression: &v1beta1.WhenExpression{
+				Input:    "foo",
+				Operator: selection.NotIn,
+				Values:   []string{"foo", "bar"},
+			},
+			IsTrue: false,
+		}},
+		Executed: false,
+	}
+
+	expectedTaskRunsStatus := make(map[string]*v1beta1.PipelineRunTaskRunStatus)
+	expectedTaskRunsStatus["test-pipeline-run-success-unit-test-1"] = &v1beta1.PipelineRunTaskRunStatus{
+		PipelineTaskName: "unit-test-1",
+		Status: &v1beta1.TaskRunStatus{
+			Status: duckv1beta1.Status{
+				Conditions: []apis.Condition{{
+					Type:    apis.ConditionSucceeded,
+					Status:  "False",
+					Reason:  "WhenExpressionsEvaluatedToFalse",
+					Message: "When Expressions for Task test-pipeline-run-success-unit-test-1 in PipelineRun test-pipeline-run evaluated to False",
+				}},
+			},
+		},
+		WhenExpressionsStatus: expectedWhenExpressionsResults,
+	}
+
+	expectedPipelineRunStatus := v1beta1.PipelineRunStatus{
+		PipelineRunStatusFields: v1beta1.PipelineRunStatusFields{
+			TaskRuns: expectedTaskRunsStatus,
+		},
+	}
+
+	pipelineRunState := resources.PipelineRunState{{
+		PipelineTask: &pipelineTask,
+		TaskRunName:  "test-pipeline-run-success-unit-test-1",
+		TaskRun:      nil,
+		ResolvedTaskResources: &taskrunresources.ResolvedTaskResources{
+			TaskSpec: &task.Spec,
+		},
+	}}
+	pr.Status.InitializeConditions()
+	status := getTaskRunsStatus(pr, pipelineRunState)
```

Comment:
hmmm interesting that you're testing getTaskRunsStatus directly (cuz it's unexported)

in this case i wonder about calling Reconcile explicitly in the test?